types.d.ts

```diff
@@ -1,16 +1,84 @@
-import { SeriesOptions, PointOptionsObject, Point, Series, SVGElement } from "highcharts";
+import { PointOptionsObject } from "highcharts";
+import Series from 'highcharts/ts/Core/Series/Series';
+import LinePoint from 'highcharts/ts/Series/Line/LinePoint';
+import SVGElement from 'highcharts/ts/Core/Renderer/SVG/SVGElement';
+import type SVGPath from "highcharts/ts/Core/Renderer/SVG/SVGPath";
+import type ColorType from "highcharts/ts/Core/Color/ColorType";
+import type DataLabelOptions from 'highcharts/ts/Core/Series/DataLabelOptions';
+import type AnimationOptions from 'highcharts/ts/Core/Animation/AnimationOptions';
+import type StatesOptions, {
+    StateGenericOptions
+} from 'highcharts/ts/Core/Series/StatesOptions';
+import type SeriesOptions, {
+    SeriesStateHoverOptions,
+    SeriesStateInactiveOptions,
+    SeriesStateNormalOptions,
+    SeriesStateSelectOptions
+} from 'highcharts/ts/Core/Series/SeriesOptions';
 
 /**
- * Shared types.
+ *
+ * Shared types
+ * 
  */
+
+interface SeriesColored extends Omit<Series, 'graph'> {
```

Comment:
This can be removed after tests are fixed, yes?

Reply:
Unfortunately, I think we need to keep this, it's used only in the type guard. Despite registering a new series, I don't see any object available on the public Highcharts we can use to check for a type.

Reply:
What about the class `ColoredlineSeries`?

Reply:
I've exported the `ColoredlineSeries` from the core code (excluding it from the compiled version), then slightly modified it to match the HC output types (generated chart data), and it seems to be working fine. Thanks!